chains/celo/listener.go

```diff
@@ -33,3 +81,32 @@ func (l *listener) start() error {
 func (l *listener) close() {
 	l.conn.Close()
 }
+
+func (l *listener) getTransactionBlockHash(hash common.Hash) (blockHash common.Hash) {
+	tx, _, err := l.conn.Client().TransactionByHash(context.Background(), hash)
+	if err != nil {
+		fmt.Errorf("unable to get BlockHash: %s", err)
+	}
+
+	receipt, err := WaitForTx(l.conn.Client(), tx)
```

Comment:
You don't need to wait here, we can assume the tx has already been processed. You should just query the receipt through the client